logging/bin/deploy_esexporter.sh

```diff
@@ -0,0 +1,94 @@
+#! /bin/bash
+
+# Copyright © 2020, SAS Institute Inc., Cary, NC, USA.  All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+cd "$(dirname $BASH_SOURCE)/../.."
+source logging/bin/common.sh
+
+this_script=`basename "$0"`
+
+log_debug "Script [$this_script] has started [$(date)]"
+
+ELASTICSEARCH_EXPORTER_ENABLED=${ELASTICSEARCH_EXPORTER_ENABLED:-true}
+
+if [ "$ELASTICSEARCH_EXPORTER_ENABLED" != "true" ]; then
+  log_warn "Environment variable [ELASTICSEARCH_EXPORTER_ENABLED] is not set to 'true'; exiting WITHOUT deploying Elasticsearch Exporter"
+  exit
+fi
+
+set -e
+
+# check for pre-reqs
+
+# Confirm namespace exists
+if [ "$(kubectl get ns $LOG_NS -o name 2>/dev/null)" == "" ]; then
+  log_error "Namespace [$LOG_NS] does NOT exist."
+  exit 98
+fi
+
+# get credentials
+if [ "$ES_METRICGETTER_USER" == "" ] || [ "$ES_METRICGETTER_PASSWD" == "" ] ; then
```

Comment:
I wanted the identities to have names that reflected their purpose rather than any particular technology currently being used.  I didn't think "exporter" made the purpose/role of the identity clear.  It makes sense if you know that the Elasticsearch Exporter is the thing that collects metrics, but not otherwise.  If we later move to some other tool for metric collection, the "metric getter" identity would still make (some) sense.  I used similar logic (?) when deciding the the identity used by Fluent Bit was the "Log Collector" identity rather than "Fluent Bit User".  If we later support Logstash or FileBeats as the log collector component, we can continue to use the "Log Collector" identity there as well.

Reply:
How about ES_METRICS_USER, ES_METRICS_PASSWD?